flytekit/experimental/eager_function.py

```diff
@@ -296,11 +312,12 @@ def get_io(dict_like):
 @asynccontextmanager
 async def eager_context(
     fn,
-    remote: FlyteRemote,
+    remote: Optional[FlyteRemote],
```

Comment:
nit
```suggestion
    remote: Optional[FlyteRemote] = None,
```

Reply:
the default `None` value isn't necessary no? The type annotation only means that the two valid inputs are `FlyteRemote` type or `None` type.

Reply:
if we use optional, we must set the default value to None. https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
If `Optional[T]` is equivalent to `X | None` or `Union[X, None]`, would we expect to always specify a default argument?

```
def fn(arg: X | None = None): ...

def fn(arg: Union[X, None] = None): ...
```

This doesn't seem correct. I think this should be totally valid, no?

```
def fn(arg: X | None): ...

def fn(arg: Union[X, None]): ...
```

All it means is value of type `X` or `None` are valid inputs.

Reply:
<img width="795" alt="image" src="https://github.com/flyteorg/flytekit/assets/2816689/9a4cd9e0-b1e1-431a-b552-4e007cedfe75">

I think all this is saying is that if `None` is a default argument, an explicit `Optional[X]` is encouraged but not necessary.